components/discourse/discourse.app.js

```diff
@@ -0,0 +1,192 @@
+        if (!categories.length) {
+          return [];
+        }
+        const rawOptions = categories.map((c) => ({
+          label: c.name,
+          value: c.id,
+        }));
+        const options = sortBy(rawOptions, ["label"]);
+
+        return {
+          options,
+        };
+      },
+    },
+    eventTypes: {
+      type: "string[]",
+      label: "Event categories",
+      description:
+        "Which Discourse events should trigger this webhook? These map to the event types you see in the Webhook UI in your Discourse Admin settings.",
+      options() {
+        // https://github.com/discourse/discourse/blob/master/app/models/web_hook_event_type.rb
+        return [
+          { value: 1, label: "Topic Event" },
+          { value: 2, label: "Post Event" },
+          { value: 3, label: "User Event" },
+          { value: 4, label: "Group Event" },
+          { value: 5, label: "Category Event" },
+          { value: 6, label: "Tag Event" },
+          { value: 9, label: "Reviewable Event" },
+          { value: 10, label: "Notification Event" },
+          { value: 13, label: "Badge Grant Event" },
+        ];
+      },
+    },
+    // The following event names were retrieved from https://github.com/discourse/discourse/blob/master/spec/models/web_hook_spec.rb
+    postEvents: {
+      type: "string[]",
+      label: "Events", // used in the context of the Post sources, this name makes sense
+      description:
+        "The type of topic events you'd like to emit from this source",
+      options() {
+        return [
+          "post_created",
+          "post_edited",
+          "post_destroyed",
+          "post_recovered",
+        ];
+      },
+    },
+    topicEvents: {
+      type: "string[]",
+      label: "Events", // used in the context of the Topic sources, this name makes sense
+      description:
+        "The type of topic events you'd like to emit from this source",
+      options() {
+        return [
+          "topic_created",
+          "topic_destroyed",
+          "topic_recovered",
+          "topic_archived_status_updated",
+          "topic_closed_status_updated",
+          "topic_visible_status_updated",
+          "topic_edited",
+        ];
+      },
+    },
+  },
+  methods: {
+    _apiUrl() {
+      return `https://${this.$auth.domain}`;
+    },
+    _apiUsername() {
+      return this.$auth.api_username;
+    },
+    _apiKey() {
+      return this.$auth.api_key;
+    },
+    _filterOnCategories(data, categories) {
+      // If no categories were passed, the user isn't
+      // filtering on category, so we return all items
+      if (!categories.length) {
+        return data;
+      }
+
+      // No way to filter items by category via API, so we filter here
+      return data.filter(
+        (el) => el.category_id && categories.includes(el.category_id.toString())
+      );
+    },
+    async _makeRequest(opts) {
+      if (!opts.headers) {
```

Comment:
```
if (!opts.headers) opts.headers = {}
// always set Content-Type and Accept like the other headers
```

Reply:
There are no cases we need to override currently. They do allow for URL-encoded form data, as well, but I see no reason why the source needs to use that. Fixed.